Examples/ArcGISToolkitExamples/JobManagerExample.swift

```diff
@@ -0,0 +1,373 @@
+        
+        // button to resume all paused jobs
+        // use this to resume the paused jobs you have after restarting your app
+        let resumeAllPausedJobsItem = UIBarButtonItem(barButtonSystemItem: .play, target: self, action: #selector(resumeAllPausedJobs))
+        
+        // button to clear the finished jobs
+        let clearFinishedJobsItem = UIBarButtonItem(barButtonSystemItem: .trash, target: self, action: #selector(clearFinishedJobs))
+        
+        let flex = UIBarButtonItem(barButtonSystemItem: .flexibleSpace, target: nil, action: nil)
+        toolbar.items = [kickOffJobItem, flex, resumeAllPausedJobsItem, flex, clearFinishedJobsItem]
+        
+        //
+        // register for user notifications, this way we can notify user in bg when job complete
+        let notificationSettings = UIUserNotificationSettings(types: [.alert, .badge, .sound], categories: nil)
+        UIApplication.shared.registerUserNotificationSettings(notificationSettings)
+        
+        // job cell registration
+        tableView.register(JobTableViewCell.self, forCellReuseIdentifier: "JobCell")
+    }
+    
+    func resumeAllPausedJobs(){
+        JobManager.shared.resumeAllPausedJobs(statusHandler: self.jobStatusHandler, completion: self.jobCompletionHandler)
+    }
+    
+    func clearFinishedJobs(){
+        JobManager.shared.clearFinishedJobs()
+        tableView.reloadData()
+    }
+    
+    var i = 0
+    
+    func kickOffJob(){
+        
+        if (i % 2) == 0{
+            let url = URL(string: "https://sampleserver6.arcgisonline.com/arcgis/rest/services/Sync/WildfireSync/FeatureServer")!
+            generateGDB(URL: url, syncModel: .layer, extent: nil)
+        }
+        else{
+
+            let portalItem = AGSPortalItem(url: URL(string:"https://www.arcgis.com/home/item.html?id=acc027394bc84c2fb04d1ed317aac674")!)!
+            let map = AGSMap(item: portalItem)
+            // naperville
+            let env = AGSEnvelope(xMin: -9825684.031125, yMin: 5102237.935062, xMax: -9798254.961608, yMax: 5151000.725314, spatialReference: AGSSpatialReference.webMercator())
+            takeOffline(map: map, extent: env)
+        }
+        
+        i += 1
+    }
+    
+    required public init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+    }
+    
+    override init(nibName nibNameOrNil: String?, bundle nibBundleOrNil: Bundle?) {
+        super.init(nibName: nil, bundle: nil)
+    }
+    
+    override open func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return jobs.count
+    }
+    
+    override open func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCell(withIdentifier: "JobCell") as! JobTableViewCell
+        let job = jobs[(indexPath as NSIndexPath).row]
```

Comment:
Why to cast indexPath to NSIndexPath?

Reply:
it was required in previous version of swift - removed it